tests/unit/utils/test_key_value_store.py

```diff
@@ -0,0 +1,113 @@
+"""Test KVS from share.ansible_navigator."""
+
+import os
+import sys
+import types
+
+import pytest
+
+from ansible_navigator.utils.key_value_store import KeyValueStore
+
+
+def test_kvs_save_restore(empty_kvs):
+    """Test basic KVS write to disk and restore."""
+    empty_kvs["hello"] = "whooop"
+    empty_kvs["i_am_a_key"] = "and I am a value"
+    empty_kvs.close()
+
+    kvs2 = KeyValueStore(empty_kvs.path)
+    assert kvs2["hello"] == "whooop"
+    assert kvs2["i_am_a_key"] == "and I am a value"
+
+
+def test_kvs_len(kvs):
+    """Test KVS __len__ / len()"""
+    assert len(kvs) == 4
+    kvs["new_key"] = "something"
+    assert len(kvs) == 5
+
+
+def test_kvs_iterkeys(kvs):
+    """Test KVS iterkeys()"""
+    # Is it truly an iterator?
+    assert isinstance(kvs.iterkeys(), types.GeneratorType)
+
+    # Does it have all the keys we expect? (sorted because dicts don't care
+    # about order)
+    assert sorted(list(kvs.iterkeys())) == ["apple", "banana", "grape", "strawberry"]
+
+
+def test_kvs_itervalues(kvs):
+    """Test KVS itervalues()"""
+    # Is it truly an iterator?
+    assert isinstance(kvs.itervalues(), types.GeneratorType)
+
+    # Does it have all the values we expect? (sorted because dicts don't care
+    # about order)
+    assert sorted(list(kvs.itervalues())) == ["green", "red", "red", "yellow"]
+
+
+def test_kvs_iteritems(kvs):
+    """Test KVS iteritems()"""
+    # Is it truly an iterator?
+    assert isinstance(kvs.iteritems(), types.GeneratorType)
+
+    # Does it have all the items we expect? (sorted because dicts don't care
+    # about order)
+    assert sorted(list(kvs.iteritems())) == [
+        ("apple", "red"),
+        ("banana", "yellow"),
+        ("grape", "green"),
+        ("strawberry", "red"),
+    ]
+
+
+# Now things get weird.
+
+
+def test_kvs_keys(kvs):
+    """Test KVS keys()"""
+    # We get back a KVSKeysView which is just a collections.abc.KeysView in
+    # disguise.
+    keys = kvs.keys()
+    assert "banana" in keys
+    assert "foo" not in keys
+    assert len(keys) == len(kvs)
```

Comment:
Could something like this work?  Since the pattern is repeated a few times..... might be handy to have

https://gist.github.com/cidrblock/19fa73e4c9f2c00e9f8ecdc9849edce8

Reply:
I'm not sure this would clean things up much, I'd still want to keep the `len()` assert. It's asserting that iterating over the keys (resp. values, items) produces the same number of results as the number of items the KVS thinks it has in it.

And the line below is testing that we don't alter the keys (resp. values, items) in some way unexpectedly.

So I'd still want to keep these asserts, I think.

Reply:
ok......  I missed the len test wasn't testing the list.....It was still early :)